File: frontend/types/meeting.ts
export interface Meeting {
  id: number;
  meeting_code: string;
  title: string;
  host_id: number;
  status: string;
  created_at: string;
  started_at: string | null;
  ended_at: string | null;
}

export interface Participant {
  id: number;
  user_id: number;
  name: string;
  email: string;
  role: string;
  joined_at: string;
  left_at: string | null;
}

export interface MeetingDetails {
  meeting: Meeting;
  participants: Participant[];
}

File: backend/app/schemas/meeting.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class MeetingCreate(BaseModel):
    title: str


class MeetingResponse(BaseModel):
    id: int
    title: str
    meeting_code: str
    host_id: int
    status: str
    created_at: datetime

    model_config = ConfigDict(
        from_attributes=True
    )


class ParticipantResponse(BaseModel):
    id: int
    user_id: int
    name: str
    email: str
    role: str
    joined_at: datetime | None = None


class MeetingDetailsResponse(BaseModel):
    meeting: MeetingResponse
    participants: list[
        ParticipantResponse
    ]